gitfile/gitfile.go

```diff
@@ -38,7 +38,7 @@ func GetFileContents(ctx context.Context, repoUrl, filepath, ref string, callbac
 }
 
 func (g *GitFile) GetFileContents(ctx context.Context, repoUrl, filepath, ref string, callback func(url string)) (io.ReadCloser, error) {
```

Comment:
What if the `callback` belonged to just `SpiTokenFetcher`. The callback is only used for initiating the OAuth flow which is not needed in the generic case. For example `EnvVarTokenFetcher` has no need for that and other fetchers might need some other configuration altogether.

Reply:
Also, this function is blocking even though it is asynchronous in nature. What if it returned a channel and operated in a go routine instead?

Reply:
callback is provided by the client. How can it sopposed to pass it directly to the fetcher ?

Reply:
will do a separate issue for that